scalafix/rules/src/main/scala/migrate/MigrationRule.scala

```diff
@@ -0,0 +1,153 @@
+package migrate
+
+import metaconfig.Configured
+import scalafix.patch.Patch
+import scalafix.util.TokenOps
+import scalafix.v1._
+import utils.ScalaExtensions._
+
+import scala.meta._
+import scala.meta.contrib.Trivia
+import scala.meta.internal.pc.ScalafixGlobal
+import scala.meta.tokens.Token
+import scala.tools.nsc.reporters.StoreReporter
+import scala.util.{Failure, Success}
+
+class MigrationRule(global: ScalafixGlobal) extends SemanticRule("MigrationRule") {
+  override def description: String = "infer types and show synthetics"
+
+  def this() = this(null)
+
+  override def withConfiguration(config: Configuration): Configured[Rule] = {
+    if (config.scalacClasspath.isEmpty) {
+      Configured.error(s"config.scalacClasspath should not be empty")
+    } else {
+      val global = CompilerService.newGlobal(config.scalacClasspath, config.scalacOptions)
+      global match {
+        case Success(settings) => Configured.ok(new MigrationRule(new ScalafixGlobal(settings, new StoreReporter, Map())))
+        case Failure(exception) => Configured.error(exception.getMessage)
+      }
+    }
+  }
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    lazy implicit val unit = global.newCompilationUnit(doc.input.text, doc.input.syntax)
+
+    val patchForExplicitResultTypes = addExplicitResultType()
+    val patchForTypeApply = AddTypeApply()
```

Comment:
This the new code... I don't know why the entire file seems new, but it's not the case